### PR TITLE
Added LUITabbedFrame.remove().

### PR DESCRIPTION
**Pull request by Schwarzbaer**
This patch assumes that refcounting will cause the added header and frame LUIObjects to be GC'd.

### CODE REVIEW

File: Builtin/LUITabbedFrame.py [hunk 1]
```diff
@@ -51,23 +50,41 @@ def add(self, header, frame):
...
-    #        if self.current_frame == frame:
-    #            self.current_frame = None
-    #        return True
-    #    else:
-    #        return False
+    def remove(self, header):
+        idx = self._find_header_index(header)
+        self.header_bar.remove_cell(idx)
+        frame = self.header_to_frame[header]
+        frame.parent = None
```
Comment (Schwarzbaer) [inline Builtin/LUITabbedFrame.py:67]: Is this sufficient to completely detach theframe, or will some operation have to be performed on the parent?

File: Builtin/LUITabbedFrame.py [hunk 2]
```diff
@@ -51,23 +50,41 @@ def add(self, header, frame):
...
+        # This code is a mess. I blame the absence of StopIteration.
+        # It's also mostly untested.
+        idx = -1
+        cont = True
+        found = False
+        children = self.header_bar.children
+        while cont:
+            idx += 1
+            child = children.next()
+            if child is None:
```
Comment (tobspr) [inline Builtin/LUITabbedFrame.py]: Why not `for i, child in enumerate(self.header_bar.children)` ?

File: Builtin/LUITabbedFrame.py [hunk 3]
```diff
@@ -51,23 +50,41 @@ def add(self, header, frame):
...
+                cont = False
+            else:
+                for grandchild in child.children:
+                    if grandchild is None:
+                        break
+                    if grandchild == header:
```
Comment (tobspr) [inline Builtin/LUITabbedFrame.py]: I don't think a check for `None` is necessary here

File: Builtin/LUITabbedFrame.py [hunk 4]
```diff
@@ -51,23 +50,41 @@ def add(self, header, frame):
...
+                        found = True
+                        cont = False
+                        break
```
Comment (tobspr) [inline Builtin/LUITabbedFrame.py]: Whats the issue with StopIteration to not be able to use it here?

File: Builtin/LUITabbedFrame.py (hunk 1 above)
Comment (tobspr) [inline Builtin/LUITabbedFrame.py:67]: `frame.parent = None` basically calls `parent.remove_child(frame)` so this should be fine.

File: Builtin/LUITabbedFrame.py (hunk 4 above)
Comment (Schwarzbaer) [inline Builtin/LUITabbedFrame.py]: lui.LUIElementIterator never throws it, but gives back an infinite sequence of Nones instead.

File: Builtin/LUITabbedFrame.py (hunk 3 above)
Comment (Schwarzbaer) [inline Builtin/LUITabbedFrame.py]: See https://github.com/tobspr/LUI/pull/13/files#r60843114 (again)

File: Builtin/LUITabbedFrame.py (hunk 2 above)
Comment (Schwarzbaer) [inline Builtin/LUITabbedFrame.py]: See https://github.com/tobspr/LUI/pull/13/files#r60843114

**Comment by tobspr** [comment]
Thanks!

**Pull request merged**